lib/src/http/common/request.dart

```diff
@@ -363,18 +369,21 @@ abstract class CommonRequest extends Object
     BaseRequest requestClone;
     final fromClient = _wTransportClient != null;
     if (this is FormRequest) {
-      requestClone =
-          fromClient ? _wTransportClient.newFormRequest() : new FormRequest();
+      requestClone = fromClient
+          ? _wTransportClient.newFormRequest()
+          : new FormRequest(transportPlatform: _transportPlatform);
```

Comment:
This is where we actually use the transport platform that is now being passed into the request instance